nni/retiarii/strategy/evolution.py

```diff
@@ -127,7 +130,8 @@ def run(self, base_model, applied_mutators):
     def _submit_config(self, config, base_model, mutators):
         _logger.debug('Model submitted to running queue: %s', config)
         model = get_targeted_model(base_model, mutators, config)
-        submit_models(model)
+        if filter_model(self.filter, model):
```

Comment:
If a model is ignored here, should `_running_models` have this model?

Reply:
I think `_running_models` should also include the filtered out model to avoid excessive running times beyond cycles and population size.

Reply:
Then you are adding a model to `_running_models` that will never be popped out.

Reply:
I think this is correct for `on_failure="ignore"`.

For `on_failure="worst"`, we can just mark the model as "a worst model", so that the algo can somehow learn to avoid selecting such model.

Reply:
I agree with you. I have modified the code in the new commit. When the model is filtered out and `on_failure="worst"`, the model will be appended to `_running_models` and marked as `Failed`, while will not be submitted. When the model is filtered out and `on_failure="ignore"`, the module will do nothing. Is this solution feasible?